dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java

```diff
@@ -62,10 +64,19 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
                 return;
             }
 
-            request.setRequestUri(uri);
             final PageMode mode = PageMode.getWithNavigateMode(request);
             try {
-                VelocityModeHandler.modeHandler(mode, request, response).serve();
+                final String pageHtml = APILocator.getHTMLPageAssetRenderedAPI().getPageHtml(
+                        PageContextBuilder.builder()
+                                .setPageUri(uri)
+                                .setPageMode(mode)
+                                .setUser(APILocator.getLoginServiceAPI().getLoggedInUser())
```

Comment:
this is not bad, but if you have the request you can use more straight-forward solution

Reply:
done https://github.com/dotCMS/core/pull/17859/commits/2e79300b85c7d196f00505d35066c610a415f580